hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/PortablePositionFactory.java

```diff
@@ -90,16 +89,12 @@ static PortableSinglePosition empty(boolean leaf, boolean any) {
         position.len = 0;
         position.leaf = leaf;
         position.any = any;
-        potentiallyNullify(position);
+        position.nil = isEmptyNil(position);
         return position;
     }
 
-    private static void potentiallyNullify(PortableSinglePosition position) {
-        if (position.isEmpty() && !position.isLeaf()) {
-            position.nil = true;
-        } else if (position.isEmpty() && position.getIndex() >= 0) {
-            position.nil = true;
-        }
+    private static boolean isEmptyNil(PortableSinglePosition position) {
+        return position.isEmpty() && (!position.isLeaf() || position.getIndex() >= 0);
     }
```

Comment:
Eliminates potential branch miss - thx @mtopolnik 
Should give a couple of nanos :)